### PR TITLE
121 support tls

### PR DESCRIPTION
**Pull request by albertobarrila**

### CODE REVIEW

File: src/connection.ts [hunk 1]
```diff
@@ -89,29 +96,31 @@ export class Connection {
...
 
   static connect(params: ConnectionParams, logger?: Logger): Promise<Connection> {
-    return new Connection(logger ?? new NullLogger()).start(params)
+    return new Connection(logger ?? new NullLogger(), params).start()
   }
 
-  public start(params: ConnectionParams): Promise<Connection> {
-    this.registerListeners(params.listeners)
+  public start(): Promise<Connection> {
+    this.registerListeners(this.params.listeners)
```
Comment (tarzacodes) [inline src/connection.ts:103]: Would have been easier to simply add a line `const params = this.params`, makes everything else less verbose

File: src/response_decoder.ts [hunk 2]
```diff
@@ -314,15 +314,15 @@ function decodeMessageHeader(dataResponse: DataReader) {
 
   const formatCode = dataResponse.readUInt8()
   const headerLength = decodeFormatCode(dataResponse, formatCode)
-  if (!headerLength) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
+  //if (!headerLength) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
```
Comment (tarzacodes) [inline src/response_decoder.ts:318]: Why was this commented out?

Comment (albertobarrila) [reply src/response_decoder.ts:317]: I've removed it it was dead code

**Review by tarzacodes — APPROVED** [review]

File: src/connection.ts [hunk 3]
```diff
@@ -482,6 +493,12 @@ export type ListenersParams = {
   publish_error?: PublishErrorListener
 }
 
+export interface SSLConnectionParams {
+  key: string
+  cert: string
+  ca?: string
+}
```
Comment (ThomasShih) [inline src/connection.ts:500]: These Params probably should be all optional and another key value pair for isEnabled? there may a use case where we would like to use TLS but we don't need to pass a key or cert.

**Pull request merged**